From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: map220v <map220v300@gmail.com>
Date: Tue, 14 Jul 2026 06:22:32 +0000
Subject: [PATCH] ICELAND: Reserve adsp_mhi_mem and global_sync_mem Both have
 XN=2(EL1&EL0 NOEXEC) in VTTBR

(cherry picked from commit 9c81fa48c662ed59c0f9d0e1f995a39d0c554d7a)
---
 arch/arm64/boot/dts/qcom/kaanapali-oneplus-iceland.dts | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/kaanapali-oneplus-iceland.dts b/arch/arm64/boot/dts/qcom/kaanapali-oneplus-iceland.dts
index afd7a2fd9145..0848c8adddf0 100644
--- a/arch/arm64/boot/dts/qcom/kaanapali-oneplus-iceland.dts
+++ b/arch/arm64/boot/dts/qcom/kaanapali-oneplus-iceland.dts
@@ -180,6 +180,11 @@
 			no-map;
 		};
 
+		adsp_mhi_mem: adsp-mhi@82380000 {
+			reg = <0x0 0x82380000 0x0 0x20000>;
+			no-map;
+		};
+
 		soccp_sdi_mem: soccp-sdi@823a0000 {
 			reg = <0x0 0x823a0000 0x0 0x40000>;
 			no-map;
@@ -210,6 +215,11 @@
 			no-map;
 		};
 
+		global_sync_mem: global-sync@d7a00000 {
+			reg = <0x0 0xd7a00000 0x0 0x200000>;
+			no-map;
+		};
+
 		/* Merged hyp_ac, tz_reserved, xbl_sc, tz_ac, and qtee regions */
 		tz_merged_mem: tz-merged@d8000000 {
 			reg = <0x0 0xd8000000 0x0 0x800000>;
